src/test/java/com/rakovets/course/java/core/practice/string/StringUtilTest.java

```diff
@@ -0,0 +1,289 @@
+    void happyEmoji(String expectedSymbol, String str1, String previousSymbol, String modernSymbol) {
+        // GIVEN
+
+        // WHEN
+        String actualString = StringUtil.happyEmoji(str1, previousSymbol, modernSymbol);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>startEndWith() {
+        return Stream.of(
+                Arguments.of("true", "Hi", "Hi"),
+                Arguments.of("false", "hi !", " Java"),
+                Arguments.of("false", "!Hi", "java")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("startEndWith")
+    void startEndWith(boolean expectedSymbol, String text, String word) {
+        // GIVEN
+
+        // WHEN
+        boolean actualString = StringUtil.startEndWith(text, word);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>vowels() {
+        return Stream.of(
+                Arguments.of("2", "HEllo!"),
+                Arguments.of("6", "qwertyuiop[]';lkjhgfdsazxcvbnm,./"),
+                Arguments.of("5", "ololoOo")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("vowels")
+    void vowels(int expectedSymbol, String str1) {
+        // GIVEN
+
+        // WHEN
+        int actualString = StringUtil.vowels(str1);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>punctuations() {
+        return Stream.of(
+                Arguments.of("1", "Hello!"),
+                Arguments.of("2", "qwertyuiop[]';lkjhgfdsazxcvbnm,./"),
+                Arguments.of("6", ",o.l?o!!.looo")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("punctuations")
+    void punctuations(int expectedSymbol, String str1) {
+        // GIVEN
+
+        // WHEN
+        int actualString = StringUtil.punctuations(str1);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>palindrome() {
+        return Stream.of(
+                Arguments.of("true", "deed"),
+                Arguments.of("false", "deeds"),
+                Arguments.of("true", "1991")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("palindrome")
+    void palindrome(boolean expectedSymbol, String str1) {
+        // GIVEN
+
+        // WHEN
+        boolean actualString = StringUtil.palindrome(str1);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>samePartsString() {
+        return Stream.of(
+                Arguments.of(new String[] {"Hel", "lo ", "Jav", "a!"}, "Hello Java!", "3"),
+                Arguments.of(new String[] {"Hello", " Java", "!"}, "Hello Java!", "5")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("samePartsString")
+    void samePartsString(String[] expectedSymbol, String str1, int i) {
```

Comment:
"Violation: 'nameMethodTest' was not used for test-method naming
Solution: Rename method
"